packages/ember-data/lib/system/model/internal-model.js

```diff
@@ -0,0 +1,696 @@
+import merge from "ember-data/system/merge";
+import RootState from "ember-data/system/model/states";
+import createRelationshipFor from "ember-data/system/relationships/state/create";
+import Snapshot from "ember-data/system/snapshot";
+import Errors from "ember-data/system/model/errors";
+
+var Promise = Ember.RSVP.Promise;
+var get = Ember.get;
+var set = Ember.set;
+var forEach = Ember.ArrayPolyfills.forEach;
+var map = Ember.ArrayPolyfills.map;
+
+var _extractPivotNameCache = Ember.create(null);
+var _splitOnDotCache = Ember.create(null);
+
+function splitOnDot(name) {
+  return _splitOnDotCache[name] || (
+    _splitOnDotCache[name] = name.split('.')
+  );
+}
+
+function extractPivotName(name) {
+  return _extractPivotNameCache[name] || (
+    _extractPivotNameCache[name] = splitOnDot(name)[0]
+  );
+}
+
+function retrieveFromCurrentState(key) {
+  return function() {
+    return get(this.currentState, key);
+  };
+}
+
+/*
```

Comment:
Can we make this a real doc block starting with `/**` and having a `@class InternalModel` otherwise yuidoc will try to stick the `@method` docblocks below on the wrong class.